deltacat/compute/compactor/compaction_session.py

```diff
@@ -56,12 +56,47 @@ def check_preconditions(
     )
 
 
+def check_build_initial_primary_key_indices_preconditions(
+            build_initial_primary_key_index: bool,
+            source_partition_locator: PartitionLocator,
+            source_compacted_partition_locator: PartitionLocator,
+            deltacat_storage):
+    if build_initial_primary_key_index:
+        assert source_compacted_partition_locator is not None
+    assert source_partition_locator.partition_values \
+        == source_compacted_partition_locator.partition_values, \
+        "Building initial primary key indices must use the same partition values " \
+        "for the source and compacted partition locator."
+    stream_locator = source_compacted_partition_locator.stream_locator
+    namespace = stream_locator.namespace
+    table_name = stream_locator.table_name
+    table_version = stream_locator.table_version
+    partition_values = source_compacted_partition_locator.partition_values
+    deltas_list_result = deltacat_storage.list_deltas(
+        namespace,
+        table_name,
+        partition_values,
+        table_version,
+        None,
+        None,
+        False,
+    )
```

Comment:
I generally prefer named args as they are easy to read and backward compatible when you add a new param to the method. Any thoughts?

Reply:
Agreed, will change to named args